runtime/gc_vlhgc/MemorySubSpaceTarok.cpp

```diff
@@ -1199,129 +1340,68 @@ MM_MemorySubSpaceTarok::timeForHeapContract(MM_EnvironmentBase *env, MM_Allocate
+	uintptr_t contractionSize = 0;
 
 	/* If there is not enough memory to satisfy the alloc, don't contract.  If allocSize is greater than the total free memory,
 	 * the currentFree value is a large positive number (negative unsigned number calculated above).
 	 */
 	if (allocSize > getApproximateActiveFreeMemorySize()) {
-		contractionSize = 0;
-	} else {
-		UDATA currentFree = getApproximateActiveFreeMemorySize() - allocSize;
-		UDATA currentHeapSize = getActiveMemorySize();
-		UDATA maximumFreePercent =  ratioContract ? OMR_MIN(_extensions->heapFreeMinimumRatioMultiplier + 5, _extensions->heapFreeMaximumRatioMultiplier + 1) :
-													_extensions->heapFreeMaximumRatioMultiplier + 1;
-		UDATA maximumFree = (currentHeapSize / _extensions->heapFreeMaximumRatioDivisor) * maximumFreePercent;
-
-		/* Do we have more free than is desirable ? */
-		if (currentFree > maximumFree ) {
-			/* How big a heap do we need to leave maximumFreePercent free given current live data */
-			UDATA targetHeapSize = ((currentHeapSize - currentFree) / (_extensions->heapFreeMaximumRatioDivisor - maximumFreePercent))
-										 * _extensions->heapFreeMaximumRatioDivisor;
-			
-			if (currentHeapSize < targetHeapSize) {
-				/* due to rounding errors, targetHeapSize may actually be larger than currentHeapSize */
-				contractionSize = 0;
-			} else {
-				/* Calculate how much we need to contract by to get to target size.
-				 * Note: PSA code will ensure we do not drop below initial heap size
-				 */
-				contractionSize= currentHeapSize - targetHeapSize;
-							
-				Trc_MM_MemorySubSpaceTarok_calculateTargetContractSize_Event1(env->getLanguageVMThread(), contractionSize);
-				
-				/* But we don't contract too quickly or by a trivial amount */	
-				UDATA maxContract = (UDATA)(currentHeapSize * _extensions->globalMaximumContraction);
-				UDATA minContract = (UDATA)(currentHeapSize * _extensions->globalMinimumContraction);
-				UDATA contractionGranule = _extensions->regionSize;
-				
-				/* If max contraction is less than a single region (minimum contraction granularity) round it up */
-				if (maxContract < contractionGranule) {
-					maxContract = contractionGranule;
-				} else {
-					maxContract = MM_Math::roundToCeiling(contractionGranule, maxContract);
-				}
-				
-				contractionSize = OMR_MIN(contractionSize, maxContract);
-				
-				/* We will contract in multiples of region size. Result may become zero */
-				contractionSize = MM_Math::roundToFloor(contractionGranule, contractionSize);
-				
-				/* Make sure contract is worthwhile, don't want to go to possible expense of a 
-				 * compact for a small contraction
-				 */
-				if (contractionSize < minContract) { 
-					contractionSize = 0;
-				}	
-				
-				Trc_MM_MemorySubSpaceTarok_calculateTargetContractSize_Event2(env->getLanguageVMThread(), contractionSize, maxContract);
-			}
-		} else {
-			/* No need to contract as current free less than max */
+		return 0;
```

Comment:
Looks like it is easy to keep single return in this function (as recommended), just put remainder if the code to `else`.